docs: update docs (#150)
nervosnetwork/light-godwoken: docs/introduction.md

    participant C as CKB indexer
    A->>B: Deposit request
    B-->>A: OK
    B->>B: Wait For deposit tx to be confirmed
    loop Every 10 seconds
        A->>C: Collect deposit cells
        C->>B: Collect cells
        B-->>C: Deposit cells
        C-->>A: Deposit cells
        note right of A: break when depsit cell is transferred to custodian lock
    end
```

#### How does deposit work

##### How to make a deposit request

All you need to do is to forge a tx to transfer assets to this deposit lock:

```json5
{
  "code_hash": <deposit lock code_hash>
  "hash_type": <deposit lock hash_type>
  "args": <Your deposit lock args>
}
```

The deposit lock args structure is declared [here](https://github.com/nervosnetwork/godwoken/blob/develop/crates/types/schemas/godwoken.mol#L175-L183) as:

```json5
// --- deposit lock ---
// a rollup_type_hash exists before this args, to make args friendly to prefix search
table DepositLockArgs {
    // layer1 lock hash
    owner_lock_hash: Byte32,
    layer2_lock: Script,
    cancel_timeout: Uint64, // v0/v1 has different min cancel_timeout, please refer to godwoken docs
    registry_id: Uint32, // only v1
}
```

##### How to fetch deposit result

> As illustrated in the deposit diagram above, user need to collect deposit cells from CKB indexer. When you got a deposit cell from indexer, and the deposit cell status is dead and transferred to the custodian lock, that means the deposit is successful.

Reviewer: ```suggestion
The user needs to collect deposit cells from the CKB indexer. When the user got a deposit cell from the indexer, and the deposit cell status is dead and transferred to the custodian lock, that means the deposit is successful.
```

> For terminal user, after making a deposit request and get a transaction hash, you can track the depsit result using CKB explorer:

Reviewer: ```suggestion
After making a deposit request and getting a transaction hash, the user can track the deposit result by CKB explorer:
```
